tests/components/lock/test_init.py

```diff
@@ -189,12 +182,12 @@ async def test_lock_open_with_code(hass: HomeAssistant) -> None:
     assert lock.state_attributes == {"code_format": r"^\d{4}$"}
 
     with pytest.raises(ValueError):
-        await _async_open(lock, ServiceCall(DOMAIN, SERVICE_OPEN, {}))
+        await lock.async_handle_open_service()
```

Comment:
Side note: If we would add the entity from a mock integration platform we could call the service instead and not have to change this test when we change the details.

Reply:
I know, but I did not want to change the tests to much to prove the correct working of the rework.
It could be added in a follow up PR

Reply:
https://github.com/home-assistant/core/pull/105020